sdk/core/core/src/az_http_policy_logging.c

```diff
@@ -56,7 +56,7 @@ static az_result _az_http_policy_logging_append_http_request_msg(
 
   if (request == NULL)
   {
-    AZ_RETURN_IF_FAILED(az_span_append(*ref_log_msg, AZ_SPAN_FROM_STR("NULL"), ref_log_msg));
+    AZ_RETURN_IF_FAILED(az_span_append(*ref_log_msg, AZ_SPAN_FROM_STR("NULL\0"), ref_log_msg));
```

Comment:
Is the \0 here to 0-terminate the message?
If so, I wouldn't do this. Instead, I keep all the logging code as it was, and then, just before calling the callback function, I'd append a 0 to the end of the span. If there is no room, then I'd just force the last byte of the span to be 0 and then call the callback.

Reply:
Done.